### ISSUE TITLE
🐞 `no space left on device`

### ISSUE THREAD

**Issue by gerhard** [dagger/dagger#6839]
![](https://uploads.linear.app/0b2a9228-3827-45ff-bb20-ccd34f57f694/e770f138-6267-4dcc-bf69-511c78ef2b8f/af9dc719-f9ee-4461-a04c-c75d97afec35?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzBiMmE5MjI4LTM4MjctNDVmZi1iYjIwLWNjZDM0ZjU3ZjY5NC9lNzcwZjEzOC02MjY3LTRkY2MtYmY2OS01MTFjNzhlZjJiOGYvYWY5ZGM3MTktZjllZS00NDYxLWEwNGMtYzc1ZDk3YWZlYzM1IiwiaWF0IjoxNzEyOTU4NDE0LCJleHAiOjMzMjgzNTE4NDE0fQ.aBG9q_GfrFHU2lq7QyAalulU0veO5N0lCj5-5XiHg90)

### What is the issue?

We have been getting reports of `no space left on device` errors with Dagger `v0.9.x` running on GitHub Actions runners.

If you are affected by this, please leave a comment with as much detail as you are able to share publicly. Especially interested in:

1. Which version of Dagger are you using?
2. Are you using Dagger Cloud?
3. How can we reproduce?
4. Can you provide `df -h` output?

> \[!TIP\]
> `[jlumbroso/free-disk-space](https://github.com/jlumbroso/free-disk-space)`
>
> ```yaml
> steps:
> - name: Free disk space
> uses: jlumbroso/free-disk-space@v1.3.1
> ```

This issue is mean to centralise private discussions and make it transparent to anyone that is affected by this. We will update this list as we learn more.

### Affected dagger versions

* Confirmed on `v0.9.6` by @alafanechere
* Confirmed on `v0.9.7` by @sagikazarmark
  * Temporary workaround: [https://github.com/openmeterio/openmeter/pull/662](https://github.com/openmeterio/openmeter/pull/662)
* Based on the above, we suspect that `v0.9.8 - v0.9.11` are affected too
* As far as we know, this also affects `v0.10.0` & `v0.10.1`

---

cc @jpadams @marcosnils @helderco @sipsma @jedevc

---

Capturing a screenshot of what this issue appears as in GitHub Actions Runners:

![image](https://uploads.linear.app/0b2a9228-3827-45ff-bb20-ccd34f57f694/86f74aa6-a54d-463b-92f0-7c31cf8356e2/10136bcd-69ab-49fc-bc32-2549d27e28b6?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzBiMmE5MjI4LTM4MjctNDVmZi1iYjIwLWNjZDM0ZjU3ZjY5NC84NmY3NGFhNi1hNTRkLTQ2M2ItOTJmMC03YzMxY2Y4MzU2ZTIvMTAxMzZiY2QtNjlhYi00OWZjLWJjMzItMjU0OWQyN2UyOGI2IiwiaWF0IjoxNzEyOTU4NDE0LCJleHAiOjMzMjgzNTE4NDE0fQ.0bJlE1KmiNz-rLtW61rehKshAeYX6_g8cDse66PYzC0 "=1717x")

**Comment by gerhard** [comment]
This seems helpful & worth adding in your pipelines as we are trying to troubleshoot this: https://github.com/openmeterio/openmeter/actions/runs/8155484168/job/22291093451#step:2:1

<img width="1323" alt="image" src="https://github.com/dagger/dagger/assets/3342/fed33570-c902-4c81-886b-826edb06fc24">

Thanks @sagikazarmark for the link!

**Comment by gerhard** [comment]
I have updated the issue description with the latest that we've learned.

We are still trying to reproduce. Any help will be appreciated.

**Comment by jpadams** [comment]
@gerhard @levlaz I see that @nicholasjackson is getting that error message here:
https://discord.com/channels/707636530424053791/1228036549145268295/1228233115965853777
https://dagger.cloud/jumppad-labs/traces/c1358caf1335a6e0fe0a29687d2c1160
```
Error: response from query: input: pipeline.directory.withFile.withFile.withFile.withFile.withFile.withFile.withFile.withNewFile.withFile.withFile.withNewFile.export resolve: failed to convert result: failed to copy files: userspace copy failed: write /var/lib/dagger/runc-overlayfs/snapshots/snapshots/503/fs/jumppad_0.0.0_windows_x86_64.zip: no space left on device
```

**Comment by jpadams** [comment]
Another one here:
https://discord.com/channels/707636530424053791/1155969163282628639/1225923538486169670

```
 on MacOS
I'm running into "no space left on device" errors when running some functions
I exec-d into the dagger engine container and I noticed this
/ # df -h
Filesystem                Size      Used Available Use% Mounted on
overlay                 109.7G    106.0G         0 100% /
tmpfs                    64.0M         0     64.0M   0% /dev
shm                      64.0M         0     64.0M   0% /dev/shm
/dev/vda1               109.7G    106.0G         0 100% /etc/resolv.conf
/dev/vda1               109.7G    106.0G         0 100% /etc/hostname
/dev/vda1               109.7G    106.0G         0 100% /etc/hosts
/dev/vda1               109.7G    106.0G         0 100% /var/lib/dagger
/dev/vda1               109.7G    106.0G         0 100% /etc/dnsmasq-resolv.conf
overlay                 109.7G    106.0G         0 100% /etc/resolv.conf

does this make sense?
DXT — 04/05/2024 2:50 PM
ah ok it seemed to be a local docker engine issue, I did a few prune commands and it's find (sic) now
```

**Comment by nicholasjackson** [comment]
@jpadams ty, I have done a few tests this morning and when I disable dagger cloud, everything is ok. I suspect this is due to a distributed go mod cache as that is the only cache volume I am using at that particular part of the build. I re-enabled the cache, build fails, out of space. 

I am just about to test running this with a new cache directory as I have just added go-build-cache, my gut felling is that this will work for a while until the cache volume grows too big. Is there any way to see the size of cache volumes in dagger cloud?

**Comment by levlaz** [comment]
@nicholasjackson following up here from our email convo 

Unfortunately, right now there is no great way to see the size of cache volume in dagger cloud. As you try to debug this a bit more - could you please add this step to your GHA pipeline and share any output that you can when this happens. 

```
    steps:
      - name: Free disk space
        uses: jlumbroso/free-disk-space@v1.3.1
```

Hopefully this provides some more clues for us.

**Comment by sagikazarmark** [comment]
@levlaz @nicholasjackson here are the settings we use for that action: https://github.com/openmeterio/openmeter/blob/main/.github/actions/free-disk-space/action.yaml

tl;dr some apt based action can **really** slow it down (several minutes).

**Comment by vikram-dagger** [comment]
New EC2 instance and Debian Linux AMI with Docker installed:

<img width="337" alt="image" src="https://github.com/user-attachments/assets/a1f9ca15-70b6-48cd-a4a9-46204d5aa585">

Dagger installed and first run using the command `dagger -m github.com/shykes/daggerverse/wolfi@v0.1.4 call --interactive container --packages=nosuchpackage`:

<img width="342" alt="image" src="https://github.com/user-attachments/assets/efe2e85e-bd96-4765-a86f-50ae085cd7e9">

**Comment by jedevc** [comment]
Going to dedupe this one against https://github.com/dagger/dagger/issues/6839.

**Comment by aweris** [comment]
> Going to dedupe this one against #6839.

@jedevc you added a self-link, weren't able to find other issue.

This is also happening in our K8s cluster. I just attached a 1TB SSD to the nodes, and we're getting this error after a couple of hours of running dagger engine on ci/cd

**Comment by jedevc** [comment]
Oops, I meant https://github.com/dagger/dagger/issues/7711.

**Comment by levlaz** [comment]
> > Going to dedupe this one against #6839.
> 
> @jedevc you added a self-link, weren't able to find other issue.
> 
> This is also happening in our K8s cluster. I just attached a 1TB SSD to the nodes, and we're getting this error after a couple of hours of running dagger engine on ci/cd

Thanks for the extra context, this is a high priority issue for us.
